File: edu.wpi.first.wpilib.plugins.cpp/resources/templates/examples/PacGoat/src/Commands/WaitForPressure.cpp
#include "WaitForPressure.h"

#include "Robot.h"

/*
 * All WPILib classes are in the FRC namespace. Either `using namespace frc` 
 * or frc scope (ex. `frc::RobotDrive`) are required to use WPILib functionality. 
 * For headers, use scope instead of `using namespace frc` to avoid global namespace
 * pollution. For source files, either option is viable.
 * See 
 */
using namespace frc;

WaitForPressure::WaitForPressure()
{
	Requires(Robot::pneumatics.get());
}

// Called just before this Command runs the first time
void WaitForPressure::Initialize() {}

// Called repeatedly when this Command is scheduled to run
void WaitForPressure::Execute() {}

// Make this return true when this Command no longer needs to run execute()
bool WaitForPressure::IsFinished()
{
	return Robot::pneumatics->IsPressurized();
}

// Called once after isFinished returns true
void WaitForPressure::End() {}

// Called when another command which requires one or more of the same
// subsystems is scheduled to run
void WaitForPressure::Interrupted() {}

File: edu.wpi.first.wpilib.plugins.java/resources/templates/examples/PacGoat/src/org/usfirst/frc/team190/pacgoat/commands/WaitForPressure.java
package $package.commands;

import edu.wpi.first.wpilibj.command.Command;

import $package.Robot;

/**
 * Wait until the pneumatics are fully pressurized. This command does nothing
 * and is intended to be used in command groups to wait for this condition.
 */
public class WaitForPressure extends Command {
	public WaitForPressure() {
		requires(Robot.pneumatics);
	}

	// Make this return true when this Command no longer needs to run execute()
	@Override
	protected boolean isFinished() {
		return Robot.pneumatics.isPressurized();
	}
}
